src/main/java/org/spongepowered/api/service/placeholder/PlaceholderText.java

```diff
@@ -0,0 +1,176 @@
+     * necessarily have to be the sender/invoker of a message, nor does it
+     * have to be the recipient. The source is selected by the context of
+     * builder. It is up to plugins that use such placeholders to be aware
+     * of the context of which the placeholder is used.</p>
+     *
+     * <p>If an invalid {@link MessageReceiver} is provided for the context
+     * of the placeholder, then the associated {@link PlaceholderParser} must
+     * return a {@link Text#EMPTY}.</p>
+     *
+     * @return The associated {@link MessageReceiver}, if any.
+     */
+    Optional<MessageReceiver> getAssociatedReceiver();
+
+    /**
+     * The variable string passed to this token to provide contextual
+     * information.
+     *
+     * @return The argument, if any
+     */
+    Optional<String> getArgument();
+
+    /**
+     * A builder for {@link PlaceholderText} objects.
+     */
+    interface Builder extends ResettableBuilder<PlaceholderText, Builder> {
+
+        /**
+         * Sets the token that represents a {@link PlaceholderParser} for use
+         * in this {@link PlaceholderText}.
+         *
+         * @param parser The {@link PlaceholderParser} to use
+         * @return This, for chaining
+         */
+        Builder setParser(PlaceholderParser parser);
+
+        /**
+         * Sets the {@link MessageReceiver} to use as a source of information
+         * for this {@link PlaceholderText} to the supplied {@link Player}.
+         *
+         * @param player The player to associate this text with.
+         * @return This, for chaining
+         *
+         * @see PlaceholderText#getAssociatedReceiver()
+         */
+        default Builder setAssociatedSource(Player player) {
+            UUID uuid = player.getUniqueId();
+            return setAssociatedSource(() -> Sponge.getServer().getPlayer(uuid).orElse(null));
+        }
+
+        /**
+         * Sets the {@link MessageReceiver} to use as a source of information
+         * for this {@link PlaceholderText}. If {@code null}, removes this source.
+         *
+         * @param supplier A {@link Supplier} that provides the
+         *      {@link MessageReceiver}
+         * @return This, for chaining
+         *
+         * @see PlaceholderText#getAssociatedReceiver()
+         */
+        Builder setAssociatedSource(@Nullable Supplier<MessageReceiver> supplier);
+
+        /**
+         * Sets a string that represents variables for the supplied token.
+         * The format of this argument string is dependent on the parser
+         * supplied to {@link #setParser(PlaceholderParser)} and thus is
+         * not prescribed here.
+         *
+         * @param string The argument string, may be null to reset to
+         *      the default argument string
+         * @return This, for chaining
+         *
+         * @see PlaceholderText#getArgument()
+         */
+        Builder setArgument(@Nullable String string);
```

Comment:
```suggestion
        Builder setArguments(@Nullable String string);
```
Like the above thought, probably want to update this builder function to be plural rather than singular

Reply:
As above.